LGAlertView/LGAlertViewController.m

```diff
@@ -52,14 +52,29 @@ - (nonnull instancetype)initWithAlertView:(nonnull LGAlertView *)alertView view:
 }
 
 - (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator {
-    [super viewWillTransitionToSize:size withTransitionCoordinator:coordinator];
+    if (coordinator) {
+        [super viewWillTransitionToSize:size withTransitionCoordinator:coordinator];
+    }
 
-    dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(0.0 * NSEC_PER_SEC)), dispatch_get_main_queue(), ^{
-        [UIView animateWithDuration:coordinator.transitionDuration animations:^{
-            [self setNeedsStatusBarAppearanceUpdate];
-            [self.alertView layoutValidateWithSize:size];
+    @try {
+        [[NSOperationQueue mainQueue] addOperationWithBlock:^{
+            __weak typeof(self) weakSelf = self;
+            [UIView animateWithDuration:0.5 animations:^{
```

Comment:
Why `0.5`? `0.5` is magic

Should be something like 
```objective-c
float transitionDuration = ^{ 
  if (coordinator) {
    return coordinator.transitionDuration;
  }
  return [LGAlertView appearance].animationDuration];
}();
```

Reply:
Check discussion plz https://github.com/Friend-LGA/LGAlertView/issues/62

Reply:
Interesting, added a comment in the discussion
https://github.com/Friend-LGA/LGAlertView/issues/62#issuecomment-2333932546